src/resolvers/block.ts

```diff
@@ -36,22 +38,66 @@ const blocks: IFieldResolver<any, any> = (obj, { numbers, hashes }: IBlocksArgs)
   if (numbers && hashes) {
     throw new Error('Only one of blocks or hashes should be provided.');
   }
-
   const f = v => web3.eth.getBlock(v, true);
   return Promise.all(numbers ? numbers.map(f) : hashes.map(f));
 };
 
+// Select multiple blocks.
+interface IBlocksRangeArgs {
+  numbers?: number[];
+  hashes?: string[];
+}
+
+const blocksRange: IFieldResolver<any, any> = async (obj, { numbers, hashes }: IBlocksRangeArgs) => {
+  if (numbers && hashes) {
+    throw new Error('Only one of blocks or hashes should be provided.');
+  }
+  type funcType = (hash: string, callback: (err: Error, res: BlockWithoutTransactionData) => void) => void;
+  
+  const getBlock: ((string) => Promise<BlockWithoutTransactionData>) = (hash) => {
+    return new Promise((res, rej) => {
+      web3.eth.getBlock(hash, (err, block) => {
+        if (err) {
+          rej(err);
+        } else {
+          res(block);
+        }
+      });
+    });
+  };
+
+  let start: number, end:number;
+
+  if (numbers && numbers.length == 2) {
+    [start, end] = numbers;
+  } else if (hashes && hashes.length == 2) {
+    const startBlock = await getBlock(hashes[0]);
```

Comment:
Could we fetch both blocks concurrently? For example:

```
[start, end] = await Promise.all(hashes.map(getBlock)).map(b => b.number);
```

Reply:
This errors out - "Promise.all(...).map is not a function"